UPGRADE-1.0.md

```diff
@@ -1,3 +1,7 @@
+# UPGRADE FROM 1.0.5 to 1.0.6
+
+* Scanning for `composer.json` file inside themes used to be recursive. This is now limited to the root of each theme. This has always been the intended location for this file.
```

Comment:
> This is now limited to the root of each theme. This has always been the intended location for this file.

That's true. What has not been said is that themes root have to be not more than one level below `app/themes`. Some people might have set up themes like `app/themes/vendor/package-name/composer.json` and some like `app/themes/FooBarTheme/composer.json` so for the first ones it would result in a BC break.

We might add an option to `FilesystemConfigurationSourceFactory` named `depth`, which defaults to `null` (being the current behaviour with no limit) and then setting it to `1` would restrict the depth in the recursive file locator.

Reply:
Just only one allow `app/themes/vendor/package-name/composer.json`.